### ISSUE TITLE
TSV loader completion slack message

### ISSUE THREAD

**Issue by AetherUnbound** [WordPress/openverse#1658]
## Description

Once WordPress/openverse-catalog#285 is done, it would be great to have a slack message once a TSV loading has completed for a given provider.

This could be done a couple of ways. One approach could be using the [`on_success_callback`](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/models/index.html#airflow.models.BaseOperator) parameter for the `load_from_s3` step of the DAG. The callback receives the Airflow context as arguments ([these values are made available](https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html#variables)), which can be used to build up a Slack notification that provides sufficient context.

I think it may be easier to have the Slack message as a separate task which runs only on the successful completion of the `load_from_s3` task. This would allow us to pull in [XComs](https://airflow.apache.org/docs/apache-airflow/stable/concepts/xcoms.html) from upstream tasks to provide more information in the Slack message. It would be great to have the following pieces of information as part of the message:

* Provider
* Media type
* Duration of the data pull task
* Number of records upserted

The best location for related functions is probably under a `dags/common/loader/reporting.py` (or similarly named) module. This function could then be added to the DAG using a [PythonOperator](https://airflow.apache.org/docs/apache-airflow/stable/howto/operator/python.html).

### Provider

We should be able to pass this in directly from the DAG, since we're defining it for each one here:

https://github.com/WordPress/openverse-catalog/blob/42b8d78778b242a6bd844755e33d818d6ba0d206/openverse_catalog/dags/common/dag_factory.py#L135

### Media type

This one may be a bit trickier since the ingestion DAGs can have multiple media types. Since each media type has its own loader, each one should probably have its own success message. I believe it can also be handed in by the DAG though, since we're iterating over the media types:

https://github.com/WordPress/openverse-catalog/blob/42b8d78778b242a6bd844755e33d818d6ba0d206/openverse_catalog/dags/common/dag_factory.py#L164

### Duration of the data pull task

This is a bit tougher. We can probably add a duration counter wrapper to the [data pull call step](https://github.com/WordPress/openverse-catalog/blob/42b8d78778b242a6bd844755e33d818d6ba0d206/openverse_catalog/dags/common/dag_factory.py#L78). This value could be pushed via XComs and retrieved by the slack message task. 

Airflow also records task duration in its metastore (example from the UI [here](https://airflow.apache.org/docs/apache-airflow/stable/ui.html#task-duration)), so we could also query the metastore and retrieve the information that way.

It's _possible_ that the Airflow context may provide an object or objects we can use to access the information. We may have to look into that further, as it could be the simplest way to get the duration.

### Number of records upserted

This one will be the hardest. It's possible that the [upsert command](https://github.com/WordPress/openverse-catalog/blob/42b8d78778b242a6bd844755e33d818d6ba0d206/openverse_catalog/dags/common/loader/sql.py#L247-L257) would return the number of rows affected ([since `DbApiHook::run` returns the execution results](https://github.com/apache/airflow/blob/main/airflow/hooks/dbapi.py#L209)). We'll have to do some testing to see if that's actually the case. In that case, we could just bubble up that value into the XComs and retrieve it in the message step.

Alternatively, we could have the `copy_to_s3` step report the number of lines present in the TSV as an XCom, and retrieve _that_ from the slack message.

Either would give us an accurate number of the affected records.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.
